lib/runtime/dart/_interceptors.js

```diff
@@ -391,7 +391,7 @@ dart_library.library('dart/_interceptors', null, /* Imports */[
         return new (_internal.IterableMixinWorkaround$(E))().asMapList(this);
       }
     }
-    dart.setBaseClass(JSArray, dart.global.Array);
+    dart.setBaseClass(JSArray, (typeof global == 'undefined' ? dart.global : global).Array);
```

Comment:
the idea behind `dart.global` was to be the global object (`window` in browser or `global` on node), was that not working for you? I'd like to find a better solution than expanding this in place. I bet we can do it though.